src/Kmd.Logic.Gateway.Automation/PreValidation/IValidation.cs

```diff
@@ -0,0 +1,11 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using Kmd.Logic.Gateway.Automation.Gateway;
+
+namespace Kmd.Logic.Gateway.Automation
+{
+    public interface IValidation
```

Comment:
Why do you need interface?

Reply:
All entity validation class going to implement this, Calling individual entity validation will easy later

Reply:
Yeah, but does it require interface and even `Task<bool>`? As far I can see, you do not need Task, but void,

Reply:
I am checking, is all validation sucsess. Planning to use for other entities, will see how we can improve in next PR

Reply:
But why it returns Task<bool> if it does not have anything async inside :( (based on the current work)